external/cache/zillow.py

```diff
@@ -0,0 +1,45 @@
+from ..models import ZillowHousing
+from django.utils import timezone
+import datetime
+from external.zillow import get_zillow_housing
+from streetaddress import StreetAddressParser
+
+
+def refresh_zillow_housing(location):
+    results = get_zillow_housing(
+        address=location.address,
+        city_state=f"{location.city}, {location.state}",
+        zipcode=location.zipcode,
+    )
+
+    addr_parser = StreetAddressParser()
+    for response in results:
+        loc_addr = addr_parser.parse(location.full_address)
+        response_addr = addr_parser.parse(response.address.full_address)
+        if loc_addr["street_full"] != response_addr["street_full"]:
+            continue
+        if location.city != response.address.city:
+            continue
+        if location.state != response.address.state:
+            continue
+
+        ZillowHousing.objects.create(
```

Comment:
it looks like this is only creating new zillow records unless I missed something? I think if we are truly just caching this data, then we need to delete the existing data when we determine that it needs to be refreshed.

Reply:
Ooooops I missed some important points. Thanks for pointing it out. Let me fix it

Reply:
Fixed it